### ISSUE TITLE
Select a way - relation members screen is missing information

### ISSUE THREAD

**Issue by SomeoneElseOSM** [MarcusWolschon/osmeditor4android#2909]
If I select https://www.openstreetmap.org/way/579786711 in Vespucci and look at the relation memberships I see:

    Role    Hiking route (King
    Role    Hiking route
    Role    Hiking route

It doesn't appear to be able to interact with that screen to obtain any details of which relation is which (the actual relations here are "[King Charles III England Coast Path (16312445)](https://www.openstreetmap.org/relation/16312445)", "[Yorkshire Wolds Way (78028)](https://www.openstreetmap.org/relation/78028)" and "[Centenary Way (16620)](https://www.openstreetmap.org/relation/16620)".

I'd expect some way to be able to see which relation is which, and the "add to relation" option to also show details of which relation is which - that also shows "Hiking route" twice.

This is in portrait - in landscape more is shown but "Role" still takes up half the screen and names are still truncated).

**Comment by simonpoole** [comment]
I suppose you mean this tab
![Screenshot_20250608_141727_Vespucci.jpg](https://github.com/user-attachments/assets/2baa0e6e-79c5-45e4-b425-af7454b1183b)?

There are some quite hard and annoying limitations with Androids AutoCompleteTextView that would require replacing even more than we've currently done with custom code to make more text visible (for example it gets near impossible to click the dropdown menus away if you make them wider).

But even as is you seem to be getting different results than I do. So the usual questions Android version and device, Vespucci version ;-)

**Comment by SomeoneElseOSM** [comment]
Android 14 (Android security update Sep 2024), Nokia X10, Text size medium, font size 3/7, display size 4/5.
Reducing the display size from 4 to 3 does show the first word after "Hiking Route".

Do you have any control over the whitespace after "Role"?  
Does "Member of relation" need to be restricted to one line, including no wrap within the box?
Does "Hiking Route" need to be displayed in full first?

**Comment by simonpoole** [comment]
First thing to know about Android, if you start using accessibility settings, things start glitching all over the place.

We could lock down the font size (which a lot of apps do) but we would naturally prefer to maintain the ability to change the setting.

> Do you have any control over the whitespace after "Role"?
There is only one space after the "relation type", Android is simply using some unknown criteria to truncate the line, I don't believe we can control that at all, but I need to recheck. Iin particular I can't remember right now if  marquee effects work for EditTexts, or just for TextViews, that would be a potential, but very slow, workaround. 

> Does "Member of relation" need to be restricted to one line, including no wrap within the box?
Two lines etc sounds fine but in practice having the rows vary in height is visually very annoying (and using a fixed line height then leads to a catch 22 with dynamically changing the font height).

> Does "Hiking Route" need to be displayed in full first?
No, but having the type of relation first makes just as much or as little sense as having a partial name first and not being able to read what kind of relation it is.

**Comment by simonpoole** [comment]
I've checked this against the branch that targets Android 15 and which will switch to using material 3 widgets were appropriate. The basic problem remains the same (no surprise), but there is no spinner widget for material instead they use an AutoCompleteTextView with input turned off, see https://github.com/material-components/material-components-android/blob/master/docs/components/Menu.md#non-editable-variation.  

Switching to this might allow us (due to more flexible styling) improve things without a radical redesign.
